src/us/kbase/workspace/database/provenance/Common.java

```diff
@@ -47,38 +52,53 @@ static <T> List<T> processSimpleList(final List<T> list, final String name) {
 			return Common.immutable(list);
 		}
 	}
-	
+
+	static String checkRpid(final String putativeRpid, final String name) {
+		final String rpid = requireNonNull(processString(putativeRpid), name);
+
+		final Pattern validRpidRegex = Pattern.compile("^\\w+:\\w+");
+		final Matcher m = validRpidRegex.matcher(rpid);
+		if (m.find()) {
+			return rpid;
+		}
+		throw new IllegalArgumentException(String.format(
+				"Illegal ID format for %s: \"%s\"\n" + "ID prefixes should match the pattern \"^\\w+:\\w+\"", name,
```

Comment:
The `+` is just for readability I guess?

Reply:
yes -- it did span two lines but I didn't notice that the formatter had turned it into a single line.

Reply:
I would either revert the formatter change or merge into a single string then

Reply:
:+1: